### ISSUE TITLE
DRAP cannot connect to RaspberryMatic running in a Docker container

### ISSUE THREAD

**Issue by ThomasSiegmund** [jens-maus/RaspberryMatic#1373]
Running RaspberryMatic in a standard Docker subnetwork on the host prevents a DRAP from connecting 
...

**Steps to reproduce the behavior**
1. Configure a DRAP to connect to a standalone RaspberryMatic. Make sure it can connect and the LED lights up in turquoise
2. Switch off the DRAP
3. Move your RaspberryMatic installation into a Docker container, e.g. by creating a backup and restoring from it in a freshly set up Docker installation according to [the installation guide](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI). I used the docker-compose variant of the docker setup.
4. Power up the DRAP
The DRAP goes to the usual sequence of yellow, green and turqoise lights. In the end it flashes in turquoise, indicating that it wants to connect to CCU but does not find it.

**Expected behavior**
The DRAP should discover the CCU and connect to it.

**System information:**
 - RaspberryMatic Version: 3.59.6.20210807
 - Host OS: Ubuntu 20.04.2 LTS (GNU/Linux 5.4.0-1041-raspi aarch64)
 - Used Hardware: Raspberry 4 
 - Used HomeMatic RF-Module: RPI-RF-MOD on HB-RF-USB-2

**Additional context**
The issue seems to be related to network discovery by UDP broadcasts. I noticed that RaspberryMatic running on bare metal is visible to EQ3's Netfinder program, but  RaspberryMatic running under Docker ist not. So I watched network traffic during Netfinder discovery.

In this example Netfinder is on 10.10.15.1, and two DRAPs are running on 10.10.15.100 und 10.10.15.101

```
No.	Time	Source	Destination	Protocol	Length	Info
4	0.648462575	10.10.15.1	255.255.255.255	UDP	52	43537 → 43439 Len=10
6	1.348330504	10.10.15.100	255.255.255.255	UDP	103	43439 → 43537 Len=61
7	1.410798515	10.10.15.101	255.255.255.255	UDP	103	43439 → 43537 Len=61
```
You can see Netfinder sending a UDP broadcast package, which is immediately answerded by the two DRAPs. I guessed that a similar mechanism might also be used by the DRAP at startup to find the CCU. However, this will fail on a standard Docker setup, because Docker does not support this kind of broadcast messages. The only easy way to fix this I know of, is to run the container in host mode like with this docker compose file:

```
version: "3.7"
services:
  raspberrymatic:
    image: ghcr.io/jens-maus/raspberrymatic:latest
    container_name: ccu
    hostname: homematic-raspi
    privileged: true
    restart: unless-stopped
    stop_grace_period: 30s
    network_mode: host
    volumes:
      - ccu_data:/usr/local:rw
      - /lib/modules:/lib/modules:ro
```
Here `network_mode: host` makes the container use the host network without any internal network or port mapping. Just adding this line to the docker-compose.yaml and restarting the container makes the DRAP connect and to go from flashing to solid turquoise light.

Interestingly, in my experience this problem exists only for the DRAPs. HAPs when run as a range extender to a CCU seem to have a different mechanism of network discovery.

**Comment by jens-maus** [comment]
Thanks for this nice elaboration, this is highly appreciated.

After some investigation, this (as you outlined) can be quite easily solved with adding `network_mode: host` in the docker/OCI use case. And I think to "solve" it for native docker/OCI users, we simply have to tune the documentation to some extend and could even adapt the `deploy.sh` script to allow to run docker using the necessary `--host` option so that the container will use the host network accordingly.

However, the same limitation exists in the HomeAssistant addon use case in principle, but seems to be quite harder to solve, I am afraid. While in principle, the HA addon could also be switched to the same `network_mode: host` mode using a home assistant addon config setting (see https://developers.home-assistant.io/docs/add-ons/configuration/#add-on-config), this could immediately negatively affect other addons running on the same HomeAssistant host. In addition, the internal RaspberryMatic iptables firewall could then quite easily mixup/break the whole host firewall setup if the RaspberryMatic HA addon will run on the host network of HomeAssistant. AFAIK, there is unfortunately indeed no solution to get a docker container to accept UDP broadcast messages that are received by the actual docker host. Perhaps @pvizeli or some other HA dev has some other idea how to forward UDP broadcast packages to the docker containers in HA without having to set `host_network: true` in the `config.json` addon config file. Otherwise, I am currently quite puzzled how to consistently solve this for the HA addon use case.

**Comment by pvizeli** [comment]
If you know the port, then you just need to set it. If it's multicast, then it could pick-up in future by the new multicast plugin

**Comment by jens-maus** [comment]
Thanks @pvizeli for the hints. Where exactly do I have to set the multicast port? In the `config.json` of the addon? Do you have some example you can show? And when will the new multicast plugin be part of HA or does a HA user have to install it himself to get udp multicast routing into a HA addon going?

**Comment by jens-maus** [comment]
> Is it possible to test a modified config.json file with "network_mode: host" inside? I can't find a way to modify this on a Addon from a repository

Sorry, but it is not possible for a user to override the `network_mode: host` for a published add-on. I, as a add-on developer would have to do that in the `config.json` of the add-on itself. However, I actually tested this myself and just using `network_mode: host` will end up in other issues that you seeing because of the add-on generally running on the host network (e.g. the network firewall will be modified by the add-on). Thus, I am still waiting for @pvizeli to answer my question and help in getting this multicast issue somewhat solved in the near future.

**Comment by pvizeli** [comment]
You can asign the port on `options` but set it to null for the host, so people can just enable it if they enter the port manual. If it using multicast, then it should be solved with https://github.com/home-assistant/plugin-multicast/issues/17

**Comment by jens-maus** [comment]
> You can asign the port on `options` but set it to null for the host, so people can just enable it if they enter the port manual.

This is already the case, indeed. However, this alone does not seem to help in the end as this issue here shows.

> If it using multicast, then it should be solved with [home-assistant/plugin-multicast#17](https://github.com/home-assistant/plugin-multicast/issues/17)

Sorry to bother you, but where exactly does `pimd` to be installed so that the multicast routing from the host to/from the add-on container works? I am really lacking some more detailed information/know-how here to get this working with HA. As said, right now the relevant udp multicast port (43438) is already defined `null` in the `config.json` file:  https://github.com/jens-maus/RaspberryMatic/blob/master/home-assistant-addon/config.json#L78

But it seems this alone does not help in getting the multicast traffic from the HA host to the add-on container and vice versa. So any more detailed hint in getting the udp multicast traffic properly routed between the HA host and the add-on container would be highly appreciated.

**Comment by dkadioglu** [comment]
I am not sure, if the following is of help or does have any short comings, so I just would like to describe my approach and I would be happy, if it helps as well as about feedback.
I have a HMIP-HAP working with Raspberrymatic in a Docker container on a RPi4 without using network_mode: host. Instead, I am using the macvlan driver with a exclusive network for the Raspberrymatic container.

This is my docker-compose file:
```
version: "3.8"
services:
  raspberrymatic:
    image: ghcr.io/jens-maus/raspberrymatic:latest
    container_name: dkhm
    hostname: dkhm
    privileged: true
    restart: unless-stopped
    stop_grace_period: 30s
    volumes:
      - /home/alarm/dkhm_data:/usr/local:rw
      - /lib/modules:/lib/modules:ro
      - /run/udev/control:/run/udev/control
      - /mnt/storage/HMBackup:/mnt:rw
    networks:
      default:
        ipv4_address: 192.168.1.221
networks:
  default:
    external:
      name: hmnet
```

Before this works, I created the following network:
```
docker network create -d macvlan -o parent=eth0 \
  --subnet 192.168.1.0/24 \
  --gateway 192.168.1.1 \
  --ip-range 192.168.1.220/30 \
  --aux-address 'host=192.168.1.222' \
  hmnet
```

To allow communication between the host and the Raspberrymatic container, the following additional steps are necessary - as the Docker firewall configuration seems to block this, as far as I understand it:
```
ip link add hmnet-shim link eth0 type macvlan mode bridge #create a virtual network device hmnet-shim bound to eth0

ip addr add 192.168.1.222/32 dev hmnet-shim #set the above mentioned (aux-address) ip address of hmnet-shim
ip link set hmnet-shim up

ip route add 192.168.1.220/30 dev hmnet-shim protocol static #add the necessary static route between host and hmnet-shim
```

As the above four lines don't survive a reboot, I created a systemd service for this:
```
[Unit]
Description=Create hmnet device and add static route
Requires=network-online.target
After=network-online.target

[Service]
Type=oneshot
ExecStart=/bin/bash -c '/usr/bin/ip link add hmnet-shim link eth0 type macvlan mode bridge && /usr/bin/ip addr add 192.168.1.222/32 dev hmnet-shim && /usr/bin/ip link set hmnet-shim up && /usr/bin/ip route add 192.168.1.220/30 dev hmnet-shim protocol static'

[Install]
WantedBy=multi-user.target
```

At least for, the communication between Raspberrymatic and the HMIP-HAP works as it should. The turquoise led is solid and not blinking. The only thing I didn't test is, if the setup process (in german "Anlernen") between Raspberrymatic and HMIP-HAP works, as I did that with network_mode: host and I can't redo the process currently. Maybe somebody else can give feedback, if the setup process works with macvlan, too.

I hope this helps.If you've got questions, please ask and I try to follow up timely.

**Comment by jens-maus** [comment]
FYI: I just tested the issue myself again and tried to use eQ-3 NetFinder to identify a running RaspberryMatic CCU as a HomeAssistant add-on. As soon as I specify port `43438/udp` in the add-on config to be forwarded to the add-on NetFinder is perfectly able to identify the RaspberryMatic also within a HomeAssistant environment. See:

![img](https://user-images.githubusercontent.com/5887769/150968740-19f9236f-5423-4205-ae12-9abd257aee7c.jpg)

The only thing that is missing/strange is, that this virtual CCU is listed with "refer to help" in the reachability column. This might be, however, due to the fact that this virtual CCU cannot be pinged/reached directly form the system running thie eQ-3 NetFinder tool.

Regarding reachability of the virtual CCU from a HmIPW-DRAP or HmIP-HAP, the situation can be easily similar and needs to be tested. Note, however, that for the DRAP to identify/find your CCU you would have to open port `43439/udp` as well.

**Comment by THorst92** [comment]
Hi @jens-maus,

I don't know if you've received already a feedback to the reachability of the virtual CCU from a HmIPW-DRAP or HmIP-HAP. I've tried to connect a HmIP-HAP to RaspberryMatic which runs as AddOn in Home Assistant without success.

SW-Version:
- RaspberryMatic 3.61.7.20220226
- HMIP-RFUSB 4.4.16
- HMIP-HAP no new update available (checked with the HomeMatic IP App) 

In RaspberryMatic all ports are open.
The ports 9293, 43438 and 43439 are activated in the configuration of the RaspberryMatic AddOn in Home Assistant.

When I reset the HMIP-HAP twice and try to activate the pairing mode, the LED stays static blue (or yellow if I block the internet connection). In RaspberryMatic I was not able to teach in the HMIP-HAP (local and with internet connection). Also in the option "Access point with incompatible firmware" the access point was not displayed.

Do you have an idea what I can try next?

**Comment by jens-maus** [comment]
> Do you have an idea what I can try next?

Nothing really. If you read through this issue ticket closely you will find reference to a HomeAssistant related ticket (https://github.com/home-assistant/plugin-multicast/issues/17) which is related to the route cause of this issue: The HmIPW-DRAP and HMIP-HAP rely on UDP multicasting network traffic which is currently not correctly routed to an add-on in HomeAssistant. Thus, the HomeAssistant developers (e.g. @pvizeli) would have to work on by enhancing their `plugin-multicast` accordingly. That's the reason why this is currently a known limitation of using the RaspberryMatic add-on within HomeAssistant.

**Comment by firstusing** [comment]
@jens-maus thanks for your efforts.
I have rebuilt my complete smart home to get the Rasperrymatic installed as Addon within HA. I was reading slowly thru your guideline and had luck to see the issue before I did start. Nothing within my home would work without DRAP and the Second HAP. 
What can we do to trigger the issue so they will get manpower to the ticket?

Thank you!

**Comment by jens-maus** [comment]
> What can we do to trigger the issue so they will get manpower to the ticket?

If you cannot work on it yourself you can just lean back and wait until someone else is going to implement the missing piece.

**Comment by vore** [comment]
> What can we do to trigger the issue so they will get manpower to the ticket?

Hello! 
Same here.. I waiting for this in order to switch to Homeassistant. Until I can't used the DRAP I can't change over.

**Comment by unaiur** [comment]
Warning: multicast and broadcast are not the same!

With multicast, a client has to register to receive the stream of packets. Multicast uses server addresses in the 224.0.0.0-239.255.255.255 range.

With broadcast, the packet is received by all the hosts in the local network without any extra step. The destination address is the broadcast address of the current subnetwork (for example 192.168.1.255)

After reading this thread, it seems that HMIP uses broadcast, not multicast. So waiting for pmid is pointless.

**Comment by unaiur** [comment]
Therefore, we just need to use a macvlan docker network in bridge mode as described here: https://docs.docker.com/network/macvlan

**Comment by jens-maus** [comment]
> Warning: multicast and broadcast are not the same!

Of course not.

> After reading this thread, it seems that HMIP uses broadcast, not multicast. So waiting for pmid is pointless.

What exactly from this thread here makes you feel this is actually the case? In fact, I still believe the HAP/DRAP are both relying on UDP multicasting and not broadcasting communication channels and thus haveing pmid running within the HA host should actually solve this issue.

> Therefore, we just need to use a macvlan docker network in bridge mode as described here  https://docs.docker.com/network/macvlan

Well, this won't help in the HomeAssistant Add-on case, I am afraid since HA does not support to start addons with macvlan. So, yes. the suggestions by @dkadioglu could be added to the documentation so direct docker users could try to get this working the way he was proposing it. But for the HomeAssistant add-on case we need a different solution.

> Which is the same proposal that @dkadioglu did 8 months ago and all you ignored.

Well, you can also come up with an own proposal or pull request if you identified a solution for the HAP/DRAP communication issue mentioned here.

**Comment by ivo7B4** [comment]
Thank you @jens-maus for all the your efforts and your great work, highly appreciated!

I'm still getting into HA, CCU3 and Raspberrymatic as I'm preparing everything for installation.

There's an observation I have where I'm not sure whether this is related to this issue:
- The installation I'm preparing has a CCU3, HAP and 2 DRAPs (and many more devices).
- I've started with CCU3 and have integrated this in HA.
- I've then got into Raspberrymatic. I've installed this on a RaspberryPi4 with a HmIP-RFUSB. After installing the backup, all wireless devices were found quite fast. I struggled however that Raspberrymatic (not as add-on with HA, the normal Raspberrymatic installation) finds the DRAPs with the connected wired devices. I've kept it running for hours. Somtimes after a power reboot of the DRAPs, I was able to detect them. I've tried this multiple times.
- I was considering whether this is a normal issue when applying the backup. I've checked with a second CCU3 where I installed the backup. No issue at all, the DRAPs were found very fast (and also no reboot of the DRAPs was required).

I thus don't know where this issue comes from:
- Is it related to HmIP-RFUSB with RaspberryPI4? Would I not see this issue with Raspberrymatic on a CCU3?
- Something else in my installation? I would not see any other issue however as I can install the backup without any problem on another CCU3.
- Is there a dependency with this issue or is this a separate issue? I share this observation in case there is any relation to this issue (which I don't assume, but I don't know).

**Comment by campinge** [comment]
Hello all! This has been pending for a while now and I just encountered the same problem. Does anyone know if it can be solved somehow in the meantime?

**Comment by jens-maus** [comment]
As long as this ticket is open and not closed there is no other solution than not using the HA addon of RaspberryMatic if you want to connect it to a HmIPW-DRAP or HmIP-HAP.

**Comment by vore** [comment]
Is it done now? Looks good from what I see. Anybody who can test and confirm?

**Comment by jens-maus** [comment]
> Is it done now?

I don't think so. It's done when it's done, thus when this ticket is closed!

**Comment by vore** [comment]
> > Is it done now?
> 
> I don't think so. It's done when it's done, thus when this ticket is closed!

Hello! The ticket is still open.. it does not look like it is going to be closed. Should I maybe ask on the current status?

**Comment by jens-maus** [comment]
> Hello! The ticket is still open.. it does not look like it is going to be closed. Should I maybe ask on the current status?

No reason to constantly ask for the status here. As already said in october: It's done when it's done. And if there is something relevant to report it will be and if the issue is solved, the ticket will be closed. Thus, no need to further discuss on the status.

**Comment by Rainerlan** [comment]
I kind of having the same problem: The HA RaspberryMatic-addon does not find the HMIP-HAP - as I read here, because its in another subnet (The LAN-Gateway is found w/o problems however).

I thought of the following solution - But before trying this out, I wanted to check with more experienced people, if the following setup may work:
- As my HA (OS) runs in a Proxmox VM (already), I would add another LXC to install RaspberryMatic as a docker container. 
- Configure host mode for the Docker container
- Select the LXC-RaspberryMatic instance in the RaspberryMatic HA integration.

Any thoughts if this might be working?
(Don't want to risk my running setup, if it may be evident that this is not going to work as intended)

**Comment by jens-maus** [comment]
> I thought of the following solution - But before trying this out, I wanted to check with more experienced people, if the following setup may work:
> 
> * As my HA (OS) runs in a Proxmox VM (already), I would add another LXC to install RaspberryMatic as a docker container.
> * Configure host mode for the Docker container
> * Select the LXC-RaspberryMatic instance in the RaspberryMatic HA integration.
> 
> Any thoughts if this might be working? (Don't want to risk my running setup, if it may be evident that this is not going to work as intended)

This is no discussion or support fora here but a bug/issue tracker. But yes, if you install RaspberryMatic as a dedicated VM (LXC does not work) it will work because then of course no docker is involved which is the root cause of the issue that a DRAP/HAP connection does not work with HA (simply because docker containers can't route multicast UDP traffic).

**Comment by marce14-git** [comment]
Sorry in advance if my response is not appropriate here:
i got it working in a docker container in unraid via a macvlan docker network 
Has to be running in privileged mode

**Comment by jens-maus** [comment]
> Sorry in advance if my response is not appropriate here: i got it working in a docker container in unraid via a macvlan docker network Has to be running in privileged mode

Sure, there are solutions for other docker environment. But the topic of this issue ticket is more than clear: Get it working a standard HomeAssistant environment. And there, macvlan is a no go!

**Comment by lukibazuki** [comment]
So if I understand this correctly, I'll have to run raspberrymatic on a separate device (not as Add-On in HA), so I can use the HAP as a Gateway. With the HACS Integration "Homematic(IP) Local" I'll still have the possibility to control everything through HA, correct? 
Is there any workaround to get the HAP working at this moment?

**Comment by jens-maus** [comment]
Please note, that I was finally able to work on that topic again and now at least for the plain docker container / OCI use case it should be possible to use a `HmIP-HAP` or `HmIPW-DRAP` if following my updated documentation at https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI and potentially also using the new `install-docker.sh` helper script as explained there.

This script (and the docu) utilizes a `macvlan` docker network setup now like previously explained by @dkadioglu (cf. https://github.com/jens-maus/RaspberryMatic/issues/1373#issuecomment-922508274) and thus should allow the docker container to communicate via UDP multicasting to a `HmIP-HAP` or `HmIPW-DRAP` and also make it unnecessary to open dedicated ports because the container will receive a dedicated ip address from the same subnet and thus should act as a "real" RaspberryMatic CCU (also with its own firewall setup).

However, please note, that for the HomeAssistant Add-on use case the issue/limitation discussed here still applies because it is currently not possible to setup the HA add-on docker container to work on dedicated docker networks, thus not being able to use a `macvlan` docker network. This is still something the HA developers (e.g. @agners, @pvizeli) are working on.

**Comment by jens-maus** [comment]
FYI: After some more work and tests with HomeAssistant I was able to create a manual patch procedure to create the necessary `macvlan` docker bridge also within HomeAssistant and and assign the RaspberryMatic add-on to utilize this macvlan bridge, thus being directly connected to the respective LAN subnet. See here for the respective documentation on how to apply this patch procedure on your HomeAssistant / RaspberryMatic add-on installation while the HA developers are still working on directly providing this functionality for add-ons like ours here:

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch

After applying this patch procedure to a running RaspberryMatic add-on it should be indeed reachable via the assigned ip address, thus communication with a local `HmIP-HAP` or `HmIPW-DRAP` in the same LAN segment should then work as expected. Note, however, that there still exists a quite strong limitation for this patch procedure, namely the point that upon reboot of HomeAssistant or even restart of the RaspberryMatic add-on one has to manually re-apply this patch procedure.

Nevertheless, if there are some HA/RaspberryMatic users out there waiting for `HmIP-HAP`/`HmIPW-DRAP` support, feel free to test this manual patch procedure and please report back if this solves your issues until the HA developers are directly integrating a more convenient solution in a future version.

**Comment by campinge** [comment]
Very well done, Jens. I Backed up my Raspberrymatic environment and installed HomeAssistant on a fresh hard drive, enabled RPI-RF-MOD. Then added the raspberrymatic addon and restored Raspberrymatic. It restored fine but then got stuck, so I ignored that, installed the Advanced SSH console via the Store (need to remember setting a password afterwards), launched it and followed your instructions.
All went absoluteley fine and I now have a steady connection between the HA Raspberrymatic Addon and my HmIP-DRAP. Let's hope the HA folks are able to pick this up much quicker now.
Thanks again for working on it!

**Comment by dersch81** [comment]
I followed your patch instructions with macvlan and the HAP directly connected to the RaspberryMatic AddOn. Many thanks. Looking forward to have some persistent patch in the future.

**Comment by Thorsten1982** [comment]
Hello, we are currently building our family house and we will use Homematic IP Wired here. I currently own a CCU2 and use Home Assistant on a Rasperry 3. I have now read that there is the Raspberrymagic OS on which I can install the Homematic addon. Then I heard that there are currently problems with the DRAP. Now there seems to be a first solution. Now my question is to assume that I can use it in 2 to 3 months without any problems? Since I want to set up everything from scratch for the new house, I could switch directly to Raspberrymagic.

**Comment by dersch81** [comment]
Is it fixed? I have upgraded RasperryMatic to 3.69.7.20230626 and HASS to 2023.6.3 and the HmIP-HAP is reachable without repeating the MAC VLAN Hotfix.

**Comment by jens-maus** [comment]
> Is it fixed?

No, the macvlan solution is still pending / not implemented in HA. Thus, as long as this ticket/issue is open the workaround patch is still required to be executed each time the addon or HA is going to be restarted...

**Comment by dersch81** [comment]
Yes, it just was reachable for some minutes. I had to use the hotfix again

**Comment by Phil7989** [comment]
> FYI: After some more work and tests with HomeAssistant I was able to create a manual patch procedure to create the necessary `macvlan` docker bridge also within HomeAssistant and assign the RaspberryMatic add-on to use this macvlan bridge, thus being directly connected to the respective LAN subnet. See here for the respective documentation on how to apply this patch procedure on your HomeAssistant / RaspberryMatic add-on installation while the HA developers are still working on directly providing this functionality for add-ons like ours here:
> 
> https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch
> 
> After applying this patch procedure to a running RaspberryMatic add-on it should be indeed reachable via the assigned ip address, thus communication with a local `HmIP-HAP` or `HmIPW-DRAP` in the same LAN segment should then work as expected. Note, however, that there still exists a quite strong limitation for this patch procedure, namely the point that upon reboot of HomeAssistant or even restart of the RaspberryMatic add-on one has to manually re-apply this patch procedure.
> 
> Nevertheless, if there are some HA/RaspberryMatic users out there waiting for `HmIP-HAP`/`HmIPW-DRAP` support, feel free to test this manual patch procedure and please report back if this solves your issues until the HA developers are directly integrating a more convenient solution in a future version.

Hi all, i have no idea about all this linux stuff. I have execute the support patch. I get the error message „RaspberryMatic isn't running or hostname incorrect.„ Do you know Thais issue? IP ans Host are correct.
![EDA4F977-80B7-4D83-B3BA-F956AC14FA60](https://github.com/jens-maus/RaspberryMatic/assets/19151312/dbb574be-b10f-47f6-ad73-31b48cbf2842)

**Comment by jens-maus** [comment]
> Hi all, i have no idea about all this linux stuff.

Don't you think that it would then be better to simply wait until a permanent solution is available? Because you will need to perform these step *each*  time you restart RaspberryMatic or your Home Assistant host. And also note that this here is no support fora, but a bug/issue tracker!

> I have execute the support patch and get the error message „RaspberryMatic isn't running or hostname incorrect.„ Do you know this issue? IP ans Host are correct.

No the hostname you have selected does not seem to be correct (see the `e.g. XXXXXXX-raspberrymatic` pattern), see the add-on page and use the correct hostname, then it should work.

**Comment by Torben0209** [comment]
Hallo Jens, wie geht es bzgl. der permanenten Implementierung des Patches in das Add-On weiter? Hab keine Lust mehr nach jedem Neustart den Patch auszuführen :-)
Beste Grüße und vielen Dank für die tolle Arbeit!

**Comment by jens-maus** [comment]
> Hallo Jens, wie geht es bzgl. der permanenten Implementierung des Patches in das Add-On weiter? Hab keine Lust mehr nach jedem Neustart den Patch auszuführen :-) Beste Grüße und vielen Dank für die tolle Arbeit!

Zu allererst: Wenn hier primär englisch gesprochen wird bleiben wir auch bei Englisch, bitte.

Und was den Patch angeht, nun dann wende dich an die HA Entwickler. Von RaspberryMatic Seite aus gibt es hier nichts weiter zu tun bzw. kann ich nichts weiter tun. Die HA Entwickler sind gefragt die notwendige Infrastruktur in deren Addon Schnittstellen zur Verfügung zu stellen damit ein Addon eine eigene separate IP-Adresse erhalten kann und das addon eben mit mac-vlan funktionalität gestartet werden kann.

Und zu guter letzt: Das ist kein Diskussionsforum hier sondern ein Bug-Issue Tracker. Wenn du nur nachfragen willst wann es hier weitergeht bzw. wie der stand ist, dann nutze den Diskussionsteil bzw. ein Diskussionsforum.

**Comment by ThomPre99** [comment]
Hi Jens,

first of all: great work you're doing!

Is it possible to start the "patch script" in the POST section of the add-on (if configured) and to control the script using configuration parameters? This means that the Raspberrymatic add-on including HmIP-HAP support would be independent of HA development (although an HA solution would of course be better). It is logical that the start time of the addon is extended if the script is activated.

Unfortunately, my docker knowledge is not so good that I could create a solution myself.

Thanks in advance and best regards
Thomas

**Comment by jens-maus** [comment]
> Is it possible to start the "patch script" in the POST section of the add-on (if configured) and to control the script using configuration parameters? This means that the Raspberrymatic add-on including HmIP-HAP support would be independent of HA development (although an HA solution would of course be better).

No, this is unfortunately **not** possible, simply because HomeAssistant does not allow to specify such PRE or POST scripts to be defined by the addon which are then executed upon start/stop in the context of the HomeAssistant host OS. Thus, a HA addon can only modify or affect things within its own limited context (within the docker container) and not  outside of that. And that's simply because of security reasons.

So, I am sorry, but the only way to get the necessary `macvlan` based networking running for being able to communicate with a `HmIPW-DRAP` or `HmIP-HAP` from a running RaspberryMatic addon is by either manually running my mentioned patch in the context of the HomeAssistant host each time your RaspberryMatic HA addon is (re-)started or simply wait for the HA developers (@pvizeli, @agners) to actually implement the necessary changes into the HA addon infrastructure so that HA addons can employ such `macvlan` docker network setups themselves.

**Comment by caldriks** [comment]
Hello jens-maus,

thank you for your great work at first!
Im currently getting in touch with ur Software and running into several Problems, especially with the HmIP-HAP Integration.
Do you recommend to setup a standalone raspberrymatic instead of the addon installation in this case?

again, thank you for your great work!

**Comment by dn-gt** [comment]
Hello,

first of all thanks for all your efforts and for providing the patch!

> https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch

Since my Home Assistant and Homematic Installation does not have Internet yet I downloaded the Shell Script and put it to the Config Directory of my Home Assistant Installation.
From there I'm running the script using config cat patch.sh | bash

Unfortunately, the script runs through but I can't reach Raspberrymatic under the new static IP Address. Any idea what's going wrong here?

`cat patch.sh | bash
ip: can't find device ''
HomeAssistant Main Gateway (e.g. 192.168.178.1): 192.168.0.2
RaspberryMatic Add-on Hostname (e.g. 5422eb72-raspberrymatic): de838cd8-raspberrymatic
RaspberryMatic Add-on IP (e.g. 192.168.178.4): 192.168.0.3
Creating docker macvlan network
every ip-range or gateway must have a corresponding subnet
Connecting add-on to macvlan network
**Error response from daemon: network ccu not found**
Stopping add-on (addon_de838cd8_raspberrymatic)
addon_de838cd8_raspberrymatic
Starting add-on (addon_de838cd8_raspberrymatic)
addon_de838cd8_raspberrymatic`

My HomeAssistant installation is available under 192.168.0.2 - I assume this is the IP I need to provide and not the Gateway, my Router, correct? I tried this as well but without any effect. Should I worry about the highlighted error message?

**Comment by ivo7B4** [comment]
Hello dn-gt
I have user knowledge for homeassistant and the integration only, but to support that jens-maus and other developers here can focus on their great work (also thanks a lot!) and as I think I can answer this question, my comment:
Do I see correctly that you have homeassistant and raspberrymatic running on separate devices? If yes, you don't need this add-on; as you have raspberrymatic already running on another device. 
What you need is the custom homematic integration (which you actually is needed in both cases to integrate CCU3 or raspberrymatic; independent wheter you are running it separately or on the same device). And the custom_homematic integration is a great integration! With having it running on two devices (and you thus don't need this add-on), you also don't need to worry about this patch as the DRAP/HAP are reacheable from the CCU3 or raspberrymatic.

And a P.S.: I also propose to use the support forums as also jens-maus has mentioned above for such topics (Diskussionsforen).

**Comment by dn-gt** [comment]
> Hello dn-gt I have user knowledge for homeassistant and the integration only, but to support that jens-maus and other developers here can focus on their great work (also thanks a lot!) and as I think I can answer this question, my comment: Do I see correctly that you have homeassistant and raspberrymatic running on separate devices? If yes, you don't need this add-on; as you have raspberrymatic already running on another device. What you need is the custom homematic integration (which you actually is needed in both cases to integrate CCU3 or raspberrymatic; independent wheter you are running it separately or on the same device). And the custom_homematic integration is a great integration! With having it running on two devices (and you thus don't need this add-on), you also don't need to worry about this patch as the DRAP/HAP are reacheable from the CCU3 or raspberrymatic.
> 
> And a P.S.: I also propose to use the support forums as also jens-maus has mentioned above for such topics (Diskussionsforen).

Thanks for your feedback, I'm using the addon version of Raspberrymatic however and not the integration. I thought it's better and saves additional hardware (the Odroid N2+ is powerful enough). I'll try to find more adequate thread in General Discussion secition. Thanks for the hint.

**Comment by ivo7B4** [comment]
> cat patch.sh | bash

I assumed that you have it on a separate device as you've mentioned two different IP addresses. I use it on two different devices; thus I don't use the add-on, only the integration. But I still would expect the same IP address.
Add-ons in home assistant you reach typically trough "settings, add-ons", then selecting the corresponding add-on and clicking on "open web-ui" afterwards. Alternatively, you can also actived "show in sidebar". I hope this helps.

**Comment by Thorsten1982** [comment]
Hello, I was able to successfully apply the patch and learn the DRAP, DRBL and DSR. Am I correct that I now always have to call Raspberymatic using the static IP. Or can you still access the Raspberrymatic web interface via the home assistant. Does calling the static IP address also work from outside? Also, for example, via VPN or Myfritz, with the background that I use German fiber optics and therefore I don't have an IPv4 address but have to use it via IPv6.

Homeassistant was assigned an Ipv6 address by the Fritzbox but the static IP address of Rapsberymatic is only an Ipv4 address. The point is that the equipment could then be set up at work and not have to be on site. Because it is a new building.

**Comment by Thorsten1982** [comment]
> Very well done, Jens. I Backed up my Raspberrymatic environment and installed HomeAssistant on a fresh hard drive, enabled RPI-RF-MOD. Then added the raspberrymatic addon and restored Raspberrymatic. It restored fine but then got stuck afterwards, so I ignored that, installed the Advanced SSH console via the Store (need to remember setting a password afterwards), launched it and followed your instructions. All went absoluteley fine and I now have a steady connection between the HA Raspberrymatic Addon and my HmIP-DRAP. Let's hope the HA folks are able to pick this up much quicker now. Thanks again for working on it!

Hello, you say it works for you without any problems.  Do you also have a wired 6-fold button in operation?  For me, 3 different buttons are not trained, the Falmot c12 could be trained.

**Comment by jens-maus** [comment]
> Hello, you say it works for you without any problems. Do you also have a wired 6-fold button in operation? For me, 3 different buttons are not trained, the Falmot c12 could be trained.

Sorry @Thorsten1982, but please stay on topic. This is not a discussion fora here, but a bug/issue ticket system and it is about HmIPW-DRAP/HmIP-HAP integration if RaspberryMatic is used as a HA addon. Thus, if you were able to apply the mentioned patch mechanism and you were able to teach-in a HmIPW-DRAP/HmIP-HAP, there is nothing more that needs to be talked here. So if you have other questions, please walk to the discussion part in GitHub here and start a new discussion or use extenal discussion fora instead. We really need to keep this issue ticket clean from such support requests, I am afraid.

**Comment by Thorsten1982** [comment]
Yes, sorry, that's of course true. I just wanted to give feedback that the patch works for me but I can't get a wall button or RT trained and thought maybe this could be related to the patch.

**Comment by jens-maus** [comment]
> Yes, sorry, that's of course true. I just wanted to give feedback that the patch works for me but I can't get a wall button or RT trained and thought maybe this could be related to the patch.

No it is not. This is a completely different layer issue which needs to be discussed elsewhere.

**Comment by ZTHawk** [comment]
I have been struggling with this decision for some time and have now moved to a standalone version instead of HA addon.
I have now 2 VirtualBox instances. One with HA and the other with RaspberryMatic.
- Advantages (at least for me):
  - UI is faster/smoother
  - Support for Accesspoint as gateway (which is important to have everything locally and mitigate HmIP server unavailibility)
- Disadvantages:
  - Requires more resources (RAM + disk space)
  - A bit more difficult to setup (https://github.com/jens-maus/RaspberryMatic/discussions/2473) (I have only assigned 1GB RAM and 8GB disk space (only 800MB are used though))
  - Same sidebar behaviour only possible with proxy (alternatively https://github.com/jens-maus/RaspberryMatic/discussions/2033 [see my comment at the end])

**Comment by jens-maus** [comment]
FYI: Please note that discussion regarding general support for `macvlan` docker container support in HomeAssistant has actually started. See here:

https://github.com/home-assistant/architecture/discussions/1034

**Comment by ciB89** [comment]
> FYI: After some more work and tests with HomeAssistant I was able to create a manual patch procedure to create the necessary `macvlan` docker bridge also within HomeAssistant and assign the RaspberryMatic add-on to use this macvlan bridge, thus being directly connected to the respective LAN subnet. See here for the respective documentation on how to apply this patch procedure on your HomeAssistant / RaspberryMatic add-on installation while the HA developers are still working on directly providing this functionality for add-ons like ours here:
> 
> https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch
> 
> After applying this patch procedure to a running RaspberryMatic add-on it should be indeed reachable via the assigned ip address, thus communication with a local `HmIP-HAP` or `HmIPW-DRAP` in the same LAN segment should then work as expected. Note, however, that there still exists a quite strong limitation for this patch procedure, namely the point that upon reboot of HomeAssistant or even restart of the RaspberryMatic add-on one has to manually re-apply this patch procedure.
> 
> Nevertheless, if there are some HA/RaspberryMatic users out there waiting for `HmIP-HAP`/`HmIPW-DRAP` support, feel free to test this manual patch procedure and please report back if this solves your issues until the HA developers are directly integrating a more convenient solution in a future version.

I've successfully used the script and can reach raspberrymatic via the new IP. But now my connection to Home Assistant via Homematic(IP) Local is not working anymore - I obviously updated the IP in the integration, but I guess I also need to setup the callback host and port? Any help here would be appreciated.

**Comment by mcmuller** [comment]
Thanks for the patch! In my enviroment (RPi4&Rasperrymatic-Addon&2xHmIP-Hap, all latest versions), I even see persistence over a HA-reboot unlike the limitation in "4.". Let's see what happens after updating HA or Rasperrymatic.

**Comment by PantheraIgnis** [comment]
For my understanding, the issue is that we cannot receive multicast upd packages within the docker container running inside HAOS? It shouldnt be too hard to create a workaround as integration, basically an integration with very little interaction to HomeAssistant itself just reading a config for its settings and listening for multicast packages and relaying them into the container via unicast. The required few lines of code can be derived from the KNX integration i think.

**Comment by jens-maus** [comment]
> For my understanding, the issue is that we cannot receive multicast upd packages within the docker container running inside HAOS? It shouldnt be too hard to create a workaround as integration, basically an integration with very little interaction to HomeAssistant itself just reading a config for its settings and listening for multicast packages and relaying them into the container via unicast. The required few lines of code can be derived from the KNX integration i think.

Well, then please go ahead and try to solve it. IMHO the only sensible and straight forward way of solving the "multicast udp" issue with HA addons requiring multicast udp communication is to support to setup the docker containers with macvlan support as suggested here https://github.com/home-assistant/architecture/discussions/1034 Anything else would be a potential workaround only. Nicer would be to have that macvlan functionality directly integrated in HA addon support itself.

**Comment by PantheraIgnis** [comment]
Macvlan would mean that we always have to define a secondary ip address for containers requiring multicast?

**Comment by jens-maus** [comment]
> Macvlan would mean that we always have to define a secondary ip address for containers requiring multicast?

Exactly. You would have to specify/reserve an additional static IP address of your LAN network which will be assigned exclusively to the add-on container (e.g. RaspberryMatic HA Add-on) under which the container will then be accessible and thus would allow to receive/send multicast udp packages.

**Comment by PantheraIgnis** [comment]
Alright i did some research: 
1. I can access all network interfaces from the main homeassistant container, inlcuding the actual network and hassio.
2. I acquired the method the KNX integration uses to open a udp multicast socket and am able to replicate that. (xknx/io/transport/udp_transport.py)

Does RaspberryMatic send messages to subscribe to the required multicast traffic and/or is there a documentation for it?
If it does i can in theory autodetect the required multicast addresses and play proxy.

**Comment by jens-maus** [comment]
> Alright i did some research:
> 
> 1. I can access all network interfaces from the main homeassistant container, inlcuding the actual network and hassio.
> 2. I acquired the method the KNX integration uses to open a udp multicast socket and am able to replicate that. (xknx/io/transport/udp_transport.py)
> 
> Does RaspberryMatic send messages to subscribe to the required multicast traffic and/or is there a documentation for it? If it does i can in theory autodetect the required multicast addresses and play proxy.

Well, as said: Feel free to develop a working example. I myself and some others have already invested quite some time in getting this whole multicast traffic communication between the CCU and the DRAP/HAP working in a docker like environment like HA addons are using, but have failed so far. I would suggest you walk through the whole discussion line above and check links to other tickets, etc. in trying to identify potential solutions apart from the current idea to simply implement macvlan support for addons in HomeAssistant. Note especially, that there is already a `plugin-multicast` available in HomeAssistant which back then we tried to modify to get this going. However, this also didn't work out, I am afraid. So, for example, have a look at:

https://github.com/home-assistant/plugin-multicast/issues/17
https://github.com/home-assistant/plugin-multicast/pull/61
https://github.com/home-assistant/plugin-multicast/pull/61#issuecomment-1426893313

Especially the last link should get you to a rather comprehensive explanation about the multicast traffic between the CCU and DRAP/HAP to identify each-other. So if you can come up with some other solution, I am of course interested to hear how to potentially implement this without the need for macvlan support. Nevertheless, please keep in mind that I am rather seeking for a straight and user-friendly way in getting this going. So having to install a somewhat workaround-integration just to forward/proxy this specific multicast-traffic to the RaspberryMatic HA add-on container reads like one step too much at the moment.

**Comment by maddhin** [comment]
Hi, as support for RPI/RPI2 is cut with OpenCCU v3.87.6.20260313, I need to find an alternative for my RPI2 setup which includes a HmIP-HAP. My RPI5 Home Assistant would be logical choice but due to missing HmIP-HAP support no option. Do I need to buy a new RPI "just" for OpenCCU or is there hope this issue can be handled? ;) 
Cheers a bunch for digging into this (which seems to be very tricky!) and making OpenCCU available!

**Comment by maedball** [comment]
Hi, 
with the last update of OpenCCU or maybe Home assistant i am facing again the issue that my HmIPW-DRAP and my HmIP-HAP are not working anymore.
OpenCCU (as app in Homeassistant): 3.87.6.20260404
Homeassistant Core: 2026.4.3
Homeassistant Supervisor: 2026.04.0
Homeassistant OS: 17.2
Homeassistant Yellow

I am still trying to figure out in which log file i might find any hints. I someone else facing issues or might offer information in which log file i might find some information? Openccu log file has no info and in the app - log there is also no info :-(.

the only error is was able to find is this one: Flag --time has been deprecated, use --timeout instead. But this is there for a long time

What is really strange is that the test of the devices is successful but they flash in turquoise, indicating that it wants to connect to CCU but does not find it.


<img width="1206" height="703" alt="Image" src="https://github.com/user-attachments/assets/d026d746-81e9-49ac-a780-27061a3356f4" />

<img width="1192" height="705" alt="Image" src="https://github.com/user-attachments/assets/67129d7b-7ee1-4f16-82ce-07193717a038" />

**Comment by langemar** [comment]
> Hi, with the last update of OpenCCU or maybe Home assistant i am facing again the issue that my HmIPW-DRAP and my HmIP-HAP are not working anymore. OpenCCU (as app in Homeassistant): 3.87.6.20260404 Homeassistant Core: 2026.4.3 Homeassistant Supervisor: 2026.04.0 Homeassistant OS: 17.2 Homeassistant Yellow
> 
> I am still trying to figure out in which log file i might find any hints. I someone else facing issues or might offer information in which log file i might find some information? Openccu log file has no info and in the app - log there is also no info :-(.
> 
> the only error is was able to find is this one: Flag --time has been deprecated, use --timeout instead. But this is there for a long time
> 
> What is really strange is that the test of the devices is successful but they flash in turquoise, indicating that it wants to connect to CCU but does not find it.
> <img alt="Image" width="1206" height="703" src="https://private-user-images.githubusercontent.com/23019308/583745263-d026d746-81e9-49ac-a780-27061a3356f4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzcxMTQ4NjEsIm5iZiI6MTc3NzExNDU2MSwicGF0aCI6Ii8yMzAxOTMwOC81ODM3NDUyNjMtZDAyNmQ3NDYtODFlOS00OWFjLWE3ODAtMjcwNjFhMzM1NmY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDI1VDEwNTYwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2NmE5Nzg2NWI1NjY3OGQ5ZjQzZGRhMzA2MzAzM2FmYzY1OGVmMmJkZTNmOTUzZTZhMzMwYTAwYTJhZjlhZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.nOvxLysz7uHMY5TcEC4QvxW6MVQmetzFifPnRZFwbt8"> <img alt="Image" width="1192" height="705" src="https://private-user-images.githubusercontent.com/23019308/583746321-67129d7b-7ee1-4f16-82ce-07193717a038.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzcxMTQ4NjEsIm5iZiI6MTc3NzExNDU2MSwicGF0aCI6Ii8yMzAxOTMwOC81ODM3NDYzMjEtNjcxMjlkN2ItN2VlMS00ZjE2LTgyY2UtMDcxOTM3MTdhMDM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDI1VDEwNTYwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkMDM1YTY5YjgxYmIzOGQwYTg2ZmRmMzQ2Y2I0YWZkMDk3MjEyOTEzMDJhZDA4NTA1NGQ2OTg5ZjU4YTBiM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.vh78NOL3DYwjEVNEYWletsMJhHNkETBUZfIi59Y0M_M">

Hi,

have you reapplied the MAC-VLAN patch as described [here](https://github.com/OpenCCU/OpenCCU/wiki/Installation-HomeAssistant#hmiphaphmipwdrap-support-patch)?
The patch does not persist re-creating the container, which is done when updating the OpenCCU Addon (App) from HA.

Best regards,
Markus

**Comment by maedball** [comment]
Thanks for the idea :-), yes i applied it before, it is also fully reachable via the applied ip, but i get a feeling that this issue is related to the network... 
Setup of the DRAP in the openCCU is working, after a few seconds the connection breaks... Looks exactly like the ICMP issues several people are facing.

**Comment by pcoura90-beep** [comment]
> Thanks for the idea :-), yes i applied it before, it is also fully reachable via the applied ip, but i get a feeling that this issue is related to the network... Setup of the DRAP in the openCCU is working, after a few seconds the connection breaks... Looks exactly like the ICMP issues several people are facing.

@maedball: i'm seeing the Same issue since a couple of days, have you been able to resolve your issue? If yes how? What do you mean with icmp issue?

Thanks and kind regards,
Philipp

**Comment by maedball** [comment]
No, i have workaround it by spinning up an openCCU on a Raspberry Pi 5. Which works without any issues. This at least ruled out my suspicion that something in my network setup changed.

The only other clue i found was that Homeassistant Green seems to also suffer some issues right now, (i am running a Homeassistant Yellow).

Some People see Issues with the connection of a CCU + HAP / DRAP when they block ICMP (Broadcast traffic) in their Network setup, but since it is working with a OpenCCU on a raspberry this is not the case. The only thing i can think of is that the MacVlan is causing this issue -> [https://github.com/OpenCCU/OpenCCU/issues/3684] Grunes is pointing this out here. I tried to replicate his fix but it is not fixing it for me -> or i am not doing it correctly.

Interestingly the connection works for a couple of seconds after a restart of the DRAP / HAP but seems to loose the keepalive connection which as far as i understood relies on Broadcast traffic again.

**Comment by ivo7B4** [comment]
One of the easiest solutions from my point of view is to use Proxmox on a miniPC and to have openCCU and home assistant running on VMs. I have DRAP and HAP as well and works without any issues.
Even better, if ever a VM restore on another Proxmox instance (e.g. replacemement of the miniPC) would be needed, the VM keeps the same MAC address, should thus get the same IP adress from the router again and thus, also the connection between openCCU and home assistant keeps working without any further configuration change of the integration in home assistant.

**Comment by maedball** [comment]
> > Thanks for the idea :-), yes i applied it before, it is also fully reachable via the applied ip, but i get a feeling that this issue is related to the network... Setup of the DRAP in the openCCU is working, after a few seconds the connection breaks... Looks exactly like the ICMP issues several people are facing.
> 
> [@maedball](https://github.com/maedball): i'm seeing the Same issue since a couple of days, have you been able to resolve your issue? If yes how? What do you mean with icmp issue?
> 
> Thanks and kind regards, Philipp

Just forgot, there have been an OS and an OpenCCU update. This does not fix it correct?

**Comment by pcoura90-beep** [comment]
> > > Thanks for the idea :-), yes i applied it before, it is also fully reacable via the applied ip, but i get a feeling that this issue is related to the network... Setup of the DRAP in the openCCU is working, after a few seconds the connection breaks... Looks exactly like the ICMP issues several people are facing.
> > 
> > 
> > [@maedball](https://github.com/maedball): i'm seeing the Same issue since a couple of days, have you been able to resolve your issue? If yes how? What do you mean with icmp issue?
> > Thanks and kind regards, Philipp
> 
> Just forgot, there have been an OS and an OpenCCU update. This does not fix it correct?

@maedball i currently suspect OS 17.2 (https://github.com/home-assistant/operating-system/releases/tag/17.2) as the issue. As it is implementing some more security around the docker networks ("Add patches with fixes for Docker engine").

I did a little bit of analysis with Wireshark and tcpdump. In the docker container the packets from the hap where received. But on my laptop i didn't see the UDP packets from the openccu. The default route in the openccu container was pointing onto the docker network in the container. So i added a rule, for the UDP traffic. The connection now seams to be stable. Sure i have not testet reboots etc.

For now i have added the following line to the script after the restart of the docker container, not sure if it will be working, but running manually it did work. 
                docker exec "${CCU_CONTAINER_NAME}" sh -c "route add -net 224.0.0.0/8 dev eth0"

The new OS was not fixing the issue.

**Comment by maedball** [comment]
@pcoura90-beep : Thank you!!!. Looks like something that can be added to the patch file. I will experiment with it.

Are you also running a homeassistnat yellow green device or something different. From your explanation it sounds like everyone that  is running the OpenCCU + HAP / DRAP is affected.

**Comment by pcoura90-beep** [comment]
Everyone that OS running HAOS + Openccu addon and a hap / drap is affected from in my opinion.


I have an Raspberry Pi 5 + RPI-RF-MOD maybe i should change the module go away from this setup.

**Comment by maedball** [comment]
During the setup time i was also thinking of putting everything to my server in the basement but than decided that all Smarthome stuff needed a dedicated hardware. And i still think it is better this way. This is why i have this specific setup.

One more question (i am currently preparing the update for the patch. -net 224.0.0.0/8 is that the internal network or your network mask?

**Comment by jens-maus** [comment]
224.0.0.0/8 is a special multicast network mask and the HAP and DRAP require multicast UDP unfortunately. That's exactly the reason why the patch exists because in a normal docker environment multicast UDP routing is usually not possible, thus that macvlan workaround is required.

So if you guys can test if this special route is really required and works and then I can adapt the patch script if you provide a PR or patch showing where to add it.

**Comment by maedball** [comment]
It is also working for me. Took a full restart of the System, without this it would not take the changes... Maybe because i have tried way to many other things.

@jens-maus: While you are in the patch, can you please also change this line:

docker stop --time 120 "${CCU_CONTAINER_NAME}"
To
docker stop --timeout 120 "${CCU_CONTAINER_NAME}"

time is deprecated and this warning is kind of annoying :-)

@pcoura90-beep:Thank you!!!!

**Comment by maddhin** [comment]
Hi, I‘m having the same problem of integrating a HAP but went the other route and bought a RPI3(!) to be on par with the current system requirements. 

One question though: would it be possible to integrate at switch into the Home Assistant OpenCCU Integration that can be switched on and activates the required configuration for UDP? Basically automating the install (and reactivation) of the patch? That would be awesome.

**Comment by pcoura90-beep** [comment]
Hey all,

I had Yesterday some time to test and well, the addon was losing its route after a restart. I also discovered that sometimes the number of the network Interface was changing.

So i decided to rework the "solution". My ideas ist now to create a RC.local File in the Container which Sets the Route automatically during each reboot and recreate the file If the original Script ist ecexuted.

https://github.com/OpenCCU/OpenCCU/pull/3858

@maedball: Can you please test the new Script as well?

@jens-maus: i have created a pull request.

**Comment by maedball** [comment]
Works :-)

**Comment by jens-maus** [comment]
After having applied the potential fix for the patch script, can you guys please show the routing table here for comparison? Thanks.

**Comment by maedball** [comment]
10.20 is my home network
```
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         homeassistant   0.0.0.0         UG    0      0        0 eth1
10.20.0.0       *               255.255.0.0     U     0      0        0 eth0
172.30.32.0     *               255.255.254.0   U     0      0        0 eth1
224.0.0.0       *               255.0.0.0       U     0      0        0 eth0
```

**Comment by maedball** [comment]
@pcoura90-beep, @jens-maus : Would you be able to make the variables also possible to be handed over as Parameters? I would like to try an realize maddhins idea.

**Comment by jens-maus** [comment]
Thanks for the example routing table. However, after some further thoughts I have some basic idea in mind how we could potentially get this issue with the 224 multicast route solved and potentially also with the requirement of using thatnoatch after each addon restart or HA reboot/update.

My basic idea is, to try to develop helper HA Addon/app which users having a HAP/DRAP would install in parallel to the normal OpenCCU Addon and which could take care of the macvlan and multicast routing setup. So if my basic idea would work out perhaps the patch script would not be required anymore... however, just give me some time to work out this potential helper addon...

**Comment by pcoura90-beep** [comment]
I have created my own automation for it. I have created an automation which triggers, every 5 minutes and checkes in the OpenCCUs IP is avaulable (Macvlan IP). If no, then it executes a shell command. This shell command connects via SSH to the HAOS and then execute the amended macvlan script. Yes it took some time to get it implemented, but this is running fine for me. 

Here is my routing table:

Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         homeassistant   0.0.0.0         UG    0      0        0 eth1
172.30.32.0     *               255.255.254.0   U     0      0        0 eth1
192.168.178.0   *               255.255.255.0   U     0      0        0 eth0
224.0.0.0       *               255.255.255.0   U     0      0        0 eth0

**Comment by jens-maus** [comment]
As said: I have planned to generate a helper add-on/app which does exactly that. Thus, no need for a cumbersome automation ;-)

**Comment by jens-maus** [comment]
> [!IMPORTANT]
> Hi all, just wanted to let all OpenCCU HomeAssistant App/Add-on users know that I just released a dedicated OpenCCU HomeAssistant Helper-App/Add-on which every HmIP-HAP/HmIPW-DRAP user can now install from the OpenCCU HA-App Repository in parallel to the normal OpenCCU HA-App/Add-on. When kept running this Helper-App will do the same like the old (and now obsolete) `patch-ha-add-macvlan.sh` script. So any regular user of that patch script should now switch to using this "OpenCCU HAP/DRAP-Helper" HA-App:

<img width="575" height="148" alt="Image" src="https://github.com/user-attachments/assets/f7f18c3d-947e-46be-8bc2-06058d5540c0" />
